### PR TITLE
Rename window to validity window

### PR DESCRIPTION
**Pull request by bwesterb**
See #21.

I like plain *window* a bit better, but this is more explicit.

### CODE REVIEW

File: draft-davidben-tls-merkle-tree-certs.md [hunk 1]
```diff
@@ -559,38 +559,38 @@ If `n` is zero, the CA does not build a tree and the tree head is `HashEmpty()`.
...
+`tree_heads` value contains the last `validity_window_size` tree heads, starting from `batch_number`, in decreasing batch number order. That is, `tree_heads[0]` is the tree head for batch `batch_number`, `tree_heads[1]` is the tree head for `batch_number - 1`, and so on. If `batch_number < validity_window_size - 1`, any tree heads with numbers below zero are filled with `HashEmpty()`.
 
-After the CA builds the Merkle Tree for a batch, it constructs the Window structure whose `batch_number` is the number of the batch being issued. It then computes a signature over the following structure:
+After the CA builds the Merkle Tree for a batch, it constructs the ValidityWindow structure whose `batch_number` is the number of the batch being issued. It then computes a signature over the following structure:
 
 ~~~
 struct {
-    uint8 label[31] = "Merkle Tree Certificate Window\0";
+    uint8 label[32] = "Merkle Tree Crts ValidityWindow\0";
```
[the remark this answers is not in the document]
Comment (bwesterb) [reply draft-davidben-tls-merkle-tree-certs.md:585]: Yeah, it fits nicely, but would be happy with 16 byte label too.

File: draft-davidben-tls-merkle-tree-certs.md [hunk 2]
```diff
@@ -470,7 +470,7 @@ For each batch in the "issued" state, the CA maintains the following batch state
 
 * The tree head, a hash computed over this list, described in {{building-tree}}.
 
-* A window signature computed as described in {{signing}}.
+* A validity window signature computed as described in {{signing}}.
```
[the remark this answers is not in the document]
Comment (bwesterb) [reply draft-davidben-tls-merkle-tree-certs.md:473]: I like window signature better. We're not going to sign other windows, are we? So perhaps just window signature?

**Comment by bwesterb** [comment]
[ ] Squash this.

File: draft-davidben-tls-merkle-tree-certs.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (davidben) [reply draft-davidben-tls-merkle-tree-certs.md]: (Marking unresolved just because it sounds like we're still deciding the terminology.)

Yeah, "window signature" would work too. My thinking around "batch's signature" was that just that we currently define this object by saying:

> For each batch in the "issued" state, the CA maintains the following batch state:
>
> * The list of assertions certified in this batch.
>
> * The tree head, a hash computed over this list, described in {{building-tree}}.
>
> * A signature over the corresponding validity window, described in {{signing}}.

And since we're associating it with the batch, it seems plausible to say "the batch's signature".

To that end, we still have a couple instances of "validity window signature" which is now a term without a definition. Whichever we pick, I think we want to be consistent between where it's defined and used.

But if we say the batch has a "window signature" and then elsewhere we say "batch's window signature", that works too. "validity window signature" would also work, I just think it's a little wordy and hard to parse. But this is a very soft preference. I don't feel very strongly and would be happy with "validity window signature" if that's the cleanest.

[the remark this answers is not in the document]
Comment (bwesterb) [reply draft-davidben-tls-merkle-tree-certs.md]: I think the draft has many moving parts, and `validity window signature` to my ear seems easier to understand even if it's wordy, but perhaps we should just see what people that read the draft for the first time prefer.

[the remark this answers is not in the document]
Comment (bwesterb) [reply draft-davidben-tls-merkle-tree-certs.md]: (We could merge and change later again — that'll reduce merge conflicts.)

**Review by davidben — APPROVED** [review]

File: draft-davidben-tls-merkle-tree-certs.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (bwesterb) [reply draft-davidben-tls-merkle-tree-certs.md:473]: > Sorry for the churn

These small details are important, and it's not perfect yet. Let's revisit it later, despite the churn :).
